packages/inventory/src/Stocktake/ListView/CreateStocktakeButton.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { ButtonWithIcon } from '@common/components';
+import { PlusCircleIcon } from '@common/icons';
+import { useTranslation } from '@common/intl';
+import { useToggle } from '@common/hooks';
+import { useInsertStocktake } from '../api';
+import { StockItemSelectModal } from 'packages/system/src';
+
+export const CreateStocktakeButton: React.FC = () => {
+  const t = useTranslation(['distribution', 'common']);
+  const { mutate } = useInsertStocktake();
+  const modalController = useToggle();
+
+  return (
+    <>
+      {modalController.isOn && (
```

Comment:
doing this so that the modal is only mounted when requested - which means the item query is only sent when you want to create a stocktake too.. all to make things as snappy as possible